Core/Operator/Instance.cs

```diff
@@ -74,28 +74,28 @@ where typeof(IInputSlot).IsAssignableFrom(field.FieldType)
 
         public bool AddConnection(Symbol.Connection connection, int multiInputIndex)
         {
-            var (_, sourceSlot, _, targetSlot) = GetInstancesForConnection(connection);
-            if (targetSlot != null)
-            {
-                targetSlot.AddConnection(sourceSlot, multiInputIndex);
-                return true;
-            }
+            var foundInstance = TryGetInstancesForConnection(connection, out _, out var sourceSlot, out _, out var targetSlot);
 
-            return false;
+            if (!foundInstance)
+                return false;
+            
+            targetSlot.AddConnection(sourceSlot, multiInputIndex);
+            return true;
         }
 
         public void RemoveConnection(Symbol.Connection connection, int index)
         {
-            var (_, _, _, targetSlot) = GetInstancesForConnection(connection);
+            var foundInstance = TryGetInstancesForConnection(connection, out _, out _, out _, out var targetSlot);
+            if (!foundInstance)
+                return;
             targetSlot.RemoveConnection(index);
         }
 
-        private (Instance, ISlot, Instance, ISlot) GetInstancesForConnection(Symbol.Connection connection)
+        private bool TryGetInstancesForConnection(Symbol.Connection connection, out Instance sourceInstance, out ISlot sourceSlot, out Instance targetInstance, out ISlot targetSlot)
```

Comment:
a lot of `out`s, but avoids having to check for nulls or create anonymous objects. We simply return `false` if it failed, and `true` if it succeeded.

Reply:
I like that.